### PR TITLE
[Part 1] Use nameof in Socket

### PR DESCRIPTION
**Pull request by jamesqo**
This PR refactors the following places to use `nameof`:

- `TraceLoggingTypeInfo`
- `UnixFileSystem`
- `NegoState`
- `SslState`
- `Socket` (split up into 2 commits, as the file is massive)

cc @stephentoub

### CODE REVIEW

File: src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs [hunk 1]
```diff
@@ -402,11 +402,11 @@ private sealed class FileSystemEnumerable<T> : IEnumerable<T>
                 // Basic validation of the input path
                 if (userPath == null)
                 {
-                    throw new ArgumentNullException("path");
+                    throw new ArgumentNullException(nameof(userPath));
                 }
                 if (string.IsNullOrWhiteSpace(userPath))
                 {
-                    throw new ArgumentException(SR.Argument_EmptyPath, "path");
+                    throw new ArgumentException(SR.Argument_EmptyPath, nameof(userPath));
```
Comment (stephentoub) [inline src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs]: Please revert the changes to this file.  There's a reason these weren't changed previously: the public entry point uses "path", not "userPath", as the variable name.

File: src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/TraceLogging/TraceLoggingTypeInfo.cs [hunk 2]
```diff
@@ -61,7 +61,7 @@ internal TraceLoggingTypeInfo(Type dataType)
 
             if (name == null)
             {
-                throw new ArgumentNullException("eventName");
+                throw new ArgumentNullException(nameof(name));
```
Comment (stephentoub) [inline src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/TraceLogging/TraceLoggingTypeInfo.cs]: Please revert the changes to this file.

File: src/System.Net.Security/src/System/Net/SecureProtocols/NegoState.cs [hunk 3]
```diff
@@ -375,13 +375,13 @@ internal void EndProcessAuthentication(IAsyncResult result)
...
             {
-                throw new ArgumentNullException("asyncResult");
+                throw new ArgumentNullException(nameof(result));
             }
 
             LazyAsyncResult lazyResult = result as LazyAsyncResult;
             if (lazyResult == null)
             {
-                throw new ArgumentException(SR.Format(SR.net_io_async_result, result.GetType().FullName), "asyncResult");
+                throw new ArgumentException(SR.Format(SR.net_io_async_result, result.GetType().FullName), nameof(result));
```
Comment (stephentoub) [inline src/System.Net.Security/src/System/Net/SecureProtocols/NegoState.cs]: Please revert the changes to this file.

File: src/System.Net.Security/src/System/Net/SecureProtocols/SslState.cs [hunk 4]
```diff
@@ -705,13 +705,13 @@ internal void EndProcessAuthentication(IAsyncResult result)
         {
             if (result == null)
             {
-                throw new ArgumentNullException("asyncResult");
+                throw new ArgumentNullException(nameof(result));
```
Comment (stephentoub) [inline src/System.Net.Security/src/System/Net/SecureProtocols/SslState.cs]: Revert.

File: src/System.Net.Security/src/System/Net/SecureProtocols/SslState.cs [hunk 5]
```diff
@@ -705,13 +705,13 @@ internal void EndProcessAuthentication(IAsyncResult result)
...
             }
 
             LazyAsyncResult lazyResult = result as LazyAsyncResult;
             if (lazyResult == null)
             {
-                throw new ArgumentException(SR.Format(SR.net_io_async_result, result.GetType().FullName), "asyncResult");
+                throw new ArgumentException(SR.Format(SR.net_io_async_result, result.GetType().FullName), nameof(result));
```
Comment (stephentoub) [inline src/System.Net.Security/src/System/Net/SecureProtocols/SslState.cs]: Revert

**Comment by stephentoub** [comment]
Please revert all of the places where your use of nameof changes the text that's used.  The rest look fine.

**Comment by jamesqo** [comment]
@stephentoub Done, all that's left are the Socket changes.

File: src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs [hunk 6]
```diff
@@ -90,7 +90,7 @@ public Socket(AddressFamily addressFamily, SocketType socketType, ProtocolType p
             s_loggingEnabled = SocketsEventSource.Log.IsEnabled();
             if (s_loggingEnabled)
             {
-                NetEventSource.Enter(NetEventSource.ComponentType.Socket, this, "Socket", addressFamily);
+                NetEventSource.Enter(NetEventSource.ComponentType.Socket, this, nameof(Socket), addressFamily);
```
Comment (justinvp) [inline src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs]: Have we considered changing `NetEventSource.Enter` and `Exit` to use `[CallerMemberName]`?

Comment (stephentoub) [inline src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs]: Yes, there's a whole issue about overhauling how this logging is done.  That would hopefully be a piece of it.

**Comment by stephentoub** [comment]
Test Innerloop OSX Debug Build and Test please

**Comment by stephentoub** [comment]
LGTM
